bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1274,6 +1314,31 @@ void ReplicaImp::onInternalMsg(InternalMessage &&msg) {
   if (auto *ppm = std::get_if<PrePrepareMsg *>(&msg)) {
     return startConsensusProcess(*ppm, true);
   }
+
+  // Handle a the PrePrepare Carrier
+  if (auto *ppcim = std::get_if<PrePrepareCarrierInternalMsg>(&msg)) {
+    if (bftEngine::ControlStateManager::instance().getPruningProcessStatus()) {
+      LOG_INFO(GL, "Received PrePrepare Carrier internal message message while pruning, so ignoring the message");
+      delete ppcim->ppm_;
+      ppcim->ppm_ = nullptr;
+      return;
+    } else {
+      return onMessage(ppcim->ppm_);
+    }
+  }
+
+  // Handle a view change indicator and trigger view change
+  if (auto *vciim = std::get_if<ViewChangeIndicatorInternalMsg>(&msg)) {
+    if (bftEngine::ControlStateManager::instance().getPruningProcessStatus()) {
+      LOG_INFO(GL, "Received Viewchange indicator internal message message while pruning, so ignoring the message");
```

Comment:
Viewchange indicator internal message -> ViewChangeIndicatorInternalMsg

Reply:
Thanks @yuliasherman for your comment, I have incorporated the changes.